ShaderTool/Command/Map.cs

```diff
@@ -337,5 +293,127 @@ class Map {
             return SUCCESS;
         }
 
+        private static int AddTexturesToResource(Stream resourceStream, MapData mapData) {
+
+            string[] mapTextureNames = mapData.textureNames;
+
+            if (mapTextureNames == null)
+                return SUCCESS;
+
+            foreach (string mapTextureName in mapTextureNames) {
+
+                string textureFilePath = Program.ResourcesFolder + "\\" + mapTextureName + ".tgx";
+
+                if (!File.Exists(textureFilePath)) {
+                    Console.WriteLine("{0} is not a texture.", mapTextureName);
+                    return WRONG_PARAMS;
+                }
+
+                byte[] textureData = File.ReadAllBytes(textureFilePath);
+                resourceStream.Write(BitConverter.GetBytes(textureData.Length));
+                resourceStream.Write(textureData);
+
+            }
+
+            return SUCCESS;
+        }
+
+        private static int AddMaterialsToResource(Stream resourceStream, MapData mapData) {
+
+            foreach (string materialName in mapData.materialNames) {
+
+                // Material not found
+                if (!Cache.MATERIALS.ContainsKey(materialName)
+                  || Cache.MATERIALS[materialName] == null) { // Not sure if this null check is even necessary, but better safe than sorry
+
+                    Console.WriteLine("Material {0} was not found!", materialName);
+                    return WRONG_PARAMS;
+
+                }
+
+                MaterialData materialData = Cache.MATERIALS[materialName];
+
+                // Write color to file, provided as an R,G,B,A float array
+                if (materialData.color == null)
+                    for (int i = 0; i < 4; i++)
+                        resourceStream.Write(BitConverter.GetBytes(0f));
```

Comment:
The default color should be 1 so that only the texture is importent. If you use zero as default the texture will be none too because you multiply by zero.
```suggestion
                        resourceStream.Write(BitConverter.GetBytes(1f));
```